### PR TITLE
[DOC] Pretty tables for Kernel#test doc

### PR DESCRIPTION
**Pull request by BurdetteLamar**
I could not get that spacing following the tables the way I wanted them.

### CODE REVIEW

File: file.c [hunk 1]
```diff
@@ -5321,61 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *      | <tt>'='</tt> | Whether the `mtime` at `path0` is equal to that at `path1`. |
+ *      | <tt>'>'</tt> | Whether the `mtime` at `path0` is greater than that at `path1`. |
+ *
+ *  - This test operates on the content of each of the entities at `path0` and `path1`,
+ *    and returns a `true` or `false`;
+ *    returns `false` if either entity does not exist:
+ *
+ *      | Character | Test |
+ *      |:---------:|:----- |
+ *      | <tt>'-'</tt> | Whether the entities exist and have the same content. |
```
[the remark this answers is not in the document]
Comment (nobu) [reply file.c:5398]: It needs the same i-node, or hard-linked files.

File: file.c [hunk 2]
```diff
@@ -5321,61 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *      |:---------:|:----- |
+ *      | <tt>'b'</tt> | Whether the entity is a block device. |
+ *      | <tt>'c'</tt> | Whether the entity is a character device. |
+ *      | <tt>'d'</tt> | Whether the entity is a directory. |
+ *      | <tt>'e'</tt> | Whether the entity is an existing entity. |
+ *      | <tt>'f'</tt> | Whether the entity is an existing regular file. |
+ *      | <tt>'g'</tt> | Whether the entity's setgid bit is set. |
+ *      | <tt>'G'</tt> | Like <tt>'g'</tt>, but also tests whether the ownership is equal to the caller's. |
+ *      | <tt>'k'</tt> | Whether the entity's sticky bit is set. |
+ *      | <tt>'l'</tt> | Whether the entry is a symbolic link. |
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c]: Fixed.  (I hated to say 'entity' throughout, but prefer it to 'thingy.')

File: file.c [hunk 3]
```diff
@@ -5321,61 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *  Performs a test on one or both of the <i>filesystem entities</i> at the given paths
+ *  `path0` and `path1`:
+ *
+ *  - Each path `path0` or `path1` points to a file, directory, device, pipe, etc.
+ *  - Character `char` selects a specific test.
+ *
+ *  The tests:
+ *
+ *  - Each of these tests operates only on the entity at `path0`,
+ *    and returns `true` or `false`:
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c]: Fixed.

File: file.c [hunk 4]
```diff
@@ -5321,61 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *      | <tt>'c'</tt> | Whether the entity is a character device. |
+ *      | <tt>'d'</tt> | Whether the entity is a directory. |
+ *      | <tt>'e'</tt> | Whether the entity is an existing entity. |
+ *      | <tt>'f'</tt> | Whether the entity is an existing regular file. |
+ *      | <tt>'g'</tt> | Whether the entity's setgid bit is set. |
+ *      | <tt>'G'</tt> | Like <tt>'g'</tt>, but also tests whether the ownership is equal to the caller's. |
+ *      | <tt>'k'</tt> | Whether the entity's sticky bit is set. |
+ *      | <tt>'l'</tt> | Whether the entry is a symbolic link. |
+ *      | <tt>'o'</tt> | Whether the entity is owned by the caller's effective uid. |
+ *      | <tt>'O'</tt> | Like <tt>'o'</tt>, but uses the real uid/gid (not the effective uid/gid). |
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c:5353]: Fixed.

File: file.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c]: I've restored the previous term 'identical';  hope that's sufficient.

File: file.c [hunk 5]
```diff
@@ -5321,61 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *
+ *  - This test operates on the content of each of the entities at `path0` and `path1`,
+ *    and returns a `true` or `false`;
+ *    returns `false` if either entity does not exist:
+ *
+ *      | Character | Test |
+ *      |:---------:|:----- |
+ *      | <tt>'-'</tt> | Whether the entities exist and have the same content. |
+ *
+ */s
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c]: Fixed.

File: file.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (nobu) [reply file.c:5353]: I think @peterzhu2118 meant `'O'`, not `'o'`.

File: file.c [hunk 6]
```diff
@@ -5321,60 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *      | <tt>'c'</tt> | Whether the entity is a character device. |
+ *      | <tt>'d'</tt> | Whether the entity is a directory. |
+ *      | <tt>'e'</tt> | Whether the entity is an existing entity. |
+ *      | <tt>'f'</tt> | Whether the entity is an existing regular file. |
+ *      | <tt>'g'</tt> | Whether the entity's setgid bit is set. |
+ *      | <tt>'G'</tt> | Like <tt>'g'</tt>, but also tests whether the ownership is equal to the caller's. |
+ *      | <tt>'k'</tt> | Whether the entity's sticky bit is set. |
+ *      | <tt>'l'</tt> | Whether the entity is a symbolic link. |
+ *      | <tt>'o'</tt> | Whether the entity is owned by the caller's effective uid/gid. |
+ *      | <tt>'O'</tt> | Like <tt>'o'</tt>, but uses the real uid/gid (not the effective uid/gid). |
```
Comment (nobu) [inline file.c:5353]:
```suggestion
 *      | <tt>'o'</tt> | Whether the entity is owned by the caller's effective uid. |
 *      | <tt>'O'</tt> | Like <tt>'o'</tt>, but uses the real uid (not the effective uid). |
```

**Review by nobu — CHANGES_REQUESTED** [review]

File: file.c [hunk 7]
```diff
@@ -5321,60 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *
+ *      | Character | Test |
+ *      |:---------:|:----- |
+ *      | <tt>'b'</tt> | Whether the entity is a block device. |
+ *      | <tt>'c'</tt> | Whether the entity is a character device. |
+ *      | <tt>'d'</tt> | Whether the entity is a directory. |
+ *      | <tt>'e'</tt> | Whether the entity is an existing entity. |
+ *      | <tt>'f'</tt> | Whether the entity is an existing regular file. |
+ *      | <tt>'g'</tt> | Whether the entity's setgid bit is set. |
+ *      | <tt>'G'</tt> | Like <tt>'g'</tt>, but also tests whether the ownership is equal to the caller's. |
```
Comment (nobu) [inline file.c:5349]: `G` is unrelated to `g`.
```suggestion
 *      | <tt>'G'</tt> | Whether the entity's group ownership is equal to the caller's. |
```

**Review by nobu — CHANGES_REQUESTED** [review]

File: file.c (hunk 6 above)
Comment (BurdetteLamar) [reply file.c:5353]: Fixed.

File: file.c (hunk 7 above)
Comment (BurdetteLamar) [reply file.c:5349]: Fixed.

File: file.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply file.c]: Fixed (I think).

File: file.c [hunk 8]
```diff
@@ -5321,60 +5321,83 @@ test_check(int n, int argc, VALUE *argv)
...
+ *  - Each path `path0` or `path1` points to a file, directory, device, pipe, etc.
+ *  - Character `char` selects a specific test.
+ *
+ *  The tests:
+ *
+ *  - Each of these tests operates only on the entity at `path0`,
+ *    and returns `true` or `false`;
+ *    for a non-existent entity, returns `false` (does not raise exception):
+ *
+ *      | Character | Test |
```
Comment (peterzhu2118) [inline file.c:5341]: Can you format the tables using something like http://markdowntable.com/?

Comment (BurdetteLamar) [reply file.c]: Done.

It did not like the alignment chars (:), so needed post-format tune-up.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**